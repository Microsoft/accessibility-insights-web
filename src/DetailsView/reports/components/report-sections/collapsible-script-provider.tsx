// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
export class CollapsibleScriptProvider {
    private addEventListenerForCollapsibleSection = () => {
        const collapsibles = document.getElementsByClassName('collapsible-control');

        for (let index = 0; index < collapsibles.length; index++) {
            const self = collapsibles[index];
            self.addEventListener('click', () => {
                const content = self.parentElement.nextElementSibling as HTMLElement;

                const expanded = self.getAttribute('aria-expanded') === 'false' ? false : true;

                self.setAttribute('aria-expanded', !expanded + '');
                content.setAttribute('aria-hidden', expanded + '');
            });
        }
    };

    public getDefault(): string {
        return `(${String(this.addEventListenerForCollapsibleSection)})()`;
    }
}
