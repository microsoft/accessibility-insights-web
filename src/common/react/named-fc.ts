// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import * as React from 'react';

export type ReactFCWithDisplayName<P = {}> = React.FC<React.PropsWithChildren<P>> & {
    displayName: string;
};

export function NamedFC<P = {}>(
    displayName: string,
    component: React.FC<React.PropsWithChildren<P>>,
): ReactFCWithDisplayName<P> {
    component.displayName = displayName;

    return component as ReactFCWithDisplayName<P>;
}
