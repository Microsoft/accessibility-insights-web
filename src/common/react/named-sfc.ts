// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import * as React from 'react';

export type ReactSFCWithDisplayName<P = {}> = React.SFC<P> & { displayName: string };

export function NamedSFC<P = {}>(displayName: string, component: React.SFC<P>): ReactSFCWithDisplayName<P> {
    component.displayName = displayName;

    return component as ReactSFCWithDisplayName<P>;
}
