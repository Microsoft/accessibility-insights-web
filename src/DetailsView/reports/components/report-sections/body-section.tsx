// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import * as React from 'react';
import { NamedSFC } from 'common/react/named-sfc';

export const BodySection = NamedSFC('BodySection', ({ children }) => {
    return <body>{children}</body>;
});
