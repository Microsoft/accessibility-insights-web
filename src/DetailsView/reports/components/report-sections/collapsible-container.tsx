// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import * as React from 'react';

import { NamedSFC } from '../../../../common/react/named-sfc';

type HeadingProps = {
    role: string;
    'aria-level': number;
};

export type CollapsibleContainerProps = {
    summaryProps: HeadingProps;
    id: string;
    summaryContent: JSX.Element;
    detailsContent: JSX.Element;
    buttonAriaLabel: string;
};

export const CollapsibleContainer = NamedSFC<CollapsibleContainerProps>('CollapsibleContainer', props => {
    const { id, summaryContent, summaryProps, detailsContent, buttonAriaLabel } = props;

    const contentId = `details-content-${id}`;

    return (
        <div className="summary-details">
            <div className="summary-container" {...summaryProps}>
                <button className="collapsible-control" aria-expanded="false" aria-controls={contentId} aria-label={buttonAriaLabel} />
                <div>{summaryContent}</div>
            </div>
            <div id={contentId} className="details-container" aria-hidden="true">
                {detailsContent}
            </div>
        </div>
    );
});
