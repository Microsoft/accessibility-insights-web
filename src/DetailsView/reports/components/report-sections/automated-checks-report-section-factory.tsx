// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import * as React from 'react';

import { NamedSFC } from '../../../../common/react/named-sfc';
import { BodySection } from './body-section';
import { ContentContainer } from './content-container';
import { DetailsSection } from './details-section';
import { FooterSection } from './footer-section';
import { HeaderSection } from './header-section';
import { ReportSectionFactory } from './report-section-factory';
import { TitleSection } from './title-section';

const createBasicComponent = (name: string) => {
    return NamedSFC(name, () => {
        return <div id={name} />;
    });
};

const createWrappingComponent = (name: string) => {
    return NamedSFC(name, ({ children }) => {
        return <div id={name}>{children}</div>;
    });
};

const Summary = createBasicComponent('summary-section');
const ResultSection = createWrappingComponent('result-section');
const FailedInstances = createBasicComponent('failed-instances-section');
const PassedChecks = createBasicComponent('passed-checks-section');
const NotApplicableChecks = createBasicComponent('not-applicable-checks-section');

// TODO most of this sections are dummy sections, the point is to replace them as we develop them
export const AutomatedChecksReportSectionFactory: ReportSectionFactory = {
    BodySection,
    ContentContainer,
    Header: HeaderSection,
    Title: TitleSection,
    Summary,
    Details: DetailsSection,
    ResultSection,
    FailedInstances,
    PassedChecks,
    NotApplicableChecks,
    Footer: FooterSection,
};
