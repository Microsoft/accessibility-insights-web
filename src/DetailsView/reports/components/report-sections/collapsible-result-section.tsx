// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import * as React from 'react';

import { NamedSFC } from '../../../../common/react/named-sfc';
import { CollapsibleContainer } from './collapsible-container';
import { ResultSectionProps } from './result-section';
import { ResultSectionTitle } from './result-section-title';
import { RulesOnly, RulesOnlyDeps, RulesOnlyProps } from './rules-only';

export type CollapsibleResultSectionProps = RulesOnlyProps &
    ResultSectionProps & {
        containerId: string;
    };

export const CollapsibleResultSection = NamedSFC<CollapsibleResultSectionProps>('CollapsibleResultSection', props => {
    const { containerClassName, containerId } = props;

    return (
        <div className={containerClassName}>
            <CollapsibleContainer
                id={containerId}
                visibleHeadingContent={<ResultSectionTitle {...props} />}
                collapsibleContent={<RulesOnly {...props} />}
                titleHeadingLevel={2}
            />
        </div>
    );
});
