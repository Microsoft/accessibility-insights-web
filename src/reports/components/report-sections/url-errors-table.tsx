// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { NamedFC } from 'common/react/named-fc';
import * as React from 'react';
import { SummaryScanError } from '../../package/accessibilityInsightsReport';
import { NewTabLink } from 'common/components/new-tab-link';
import { SummaryResultsTable } from 'reports/components/report-sections/summary-results-table';

export type UrlErrorsTableProps = {
    errors: SummaryScanError[];
};

export const UrlErrorsTable = NamedFC<UrlErrorsTableProps>('UrlErrorsTable', props => {
    const errors = props.errors;

    const headers = ['Error type', 'URL', 'Error description'];

    const rows = errors.map(scanError => {
        const { errorDescription, errorType, url } = scanError;
        const urlLink = <NewTabLink href={url}>{url}</NewTabLink>;
        return [errorType, urlLink, errorDescription];
    });

    return <SummaryResultsTable columnHeaders={headers} rows={rows} />;
});
