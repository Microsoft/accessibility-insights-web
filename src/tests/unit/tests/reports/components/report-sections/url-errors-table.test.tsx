// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { shallow } from 'enzyme';
import * as React from 'react';
import { UrlErrorsTable } from 'reports/components/report-sections/url-errors-table';

describe(UrlErrorsTable, () => {
    it('renders', () => {
        const errors = [
            {
                url: 'https://url.com/1',
                errorType: 'type1',
                errorDescription: 'error description 1',
            },
            {
                url: 'https://url.com/2',
                errorType: 'type2',
                errorDescription: 'error description 2',
            },
            {
                url: 'https://url.com/3',
                errorType: 'type3',
                errorDescription: 'error description 3',
            },
        ];

        const wrapper = shallow(<UrlErrorsTable errors={errors} />);

        expect(wrapper.getElement()).toMatchSnapshot();
    });
});
