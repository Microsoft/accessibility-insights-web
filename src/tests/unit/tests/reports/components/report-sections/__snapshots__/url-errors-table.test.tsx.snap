// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` renders 1`] = `
<SummaryResultsTable
  columnHeaders={
    Array [
      "Error type",
      "URL",
      "Error description",
    ]
  }
  rows={
    Array [
      Array [
        "type1",
        <NewTabLink
          href="https://url.com/1"
        >
          https://url.com/1
        </NewTabLink>,
        "error description 1",
      ],
      Array [
        "type2",
        <NewTabLink
          href="https://url.com/2"
        >
          https://url.com/2
        </NewTabLink>,
        "error description 2",
      ],
      Array [
        "type3",
        <NewTabLink
          href="https://url.com/3"
        >
          https://url.com/3
        </NewTabLink>,
        "error description 3",
      ],
    ]
  }
/>
`;
